src/components/shared/SearchInput.js

```diff
@@ -141,6 +146,47 @@ class SearchInput extends Component<Props, State> {
     }
   };
 
+  onKeyDown = (e: any) => {
+    const { onHistoryScroll, onKeyDown } = this.props;
+    if (onHistoryScroll) {
```

Comment:
maybe we should invert this: `if (!onHistoryScroll) {return}` or do `if (onHist) { return onHistoryScroll(e) }`